pkg/distributor/push.go

```diff
@@ -100,14 +100,22 @@ func Handler(
 
 type distributorMaxWriteMessageSizeErr struct {
 	actual, limit int
+	id            globalerror.ID
 }
 
 func (e distributorMaxWriteMessageSizeErr) Error() string {
 	msgSizeDesc := fmt.Sprintf(" of %d bytes", e.actual)
+	var flag string
 	if e.actual < 0 {
 		msgSizeDesc = ""
 	}
-	return globalerror.DistributorMaxWriteMessageSize.MessageWithPerInstanceLimitConfig(fmt.Sprintf("the incoming push request has been rejected because its message size%s is larger than the allowed limit of %d bytes", msgSizeDesc, e.limit), "distributor.max-recv-msg-size")
+	if e.id == globalerror.DistributorMaxOtelDecompressedWriteMessageSize {
```

Comment:
Instead of having extra logic in `Error` method, can we have two separate types for each error (one for OTLP, one for Push) or put `flag` into existing type and create two separate constructors (`newPushMaxMessageSizeError` and `newOTLPMaxMessageSizeError`) for each type.

Reply:
Agree :point_up:

Reply:
was thinking about those, didn't want to duplicate the code, if we find this more clear, I am open to change.

Reply:
error split is done in commit https://github.com/grafana/mimir/commit/1ba1bcb4854d016dfada97077fc1696ea23c84e3